src/pyaro/timeseries/Filter.py

```diff
@@ -1184,3 +1184,162 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
         selected_names = names[mask]
 
         return {name: stations[name] for name in selected_names}
+
+
+@registered_filter
+class ValleyFloorRelativeAltitudeFilter(StationFilter):
+    EARTH_RADIUS = 6378137  # meters
+
+    # https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#latitude-coordinate
+    UNITS_LAT = set(
+        [
+            "degrees_north",
+            "degree_north",
+            "degree_N",
+            "degrees_N",
+            "degreeN",
+            "degreesN",
+        ]
+    )
+
+    # https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#longitude-coordinate
+    UNITS_LON = set(
+        ["degrees_east", "degree_east", "degree_E", "degrees_E", "degreeE", "degreesE"]
+    )
+
+    def __init__(
+        self,
+        topo_file: str | None = None,
+        *,
+        radius: float = 5000,
+        topo_var: str = "Band1",
+        lower: float | None = None,
+        upper: float | None = None,
+    ):
+        self._topo_file = topo_file
+        self._topo_var = topo_var
+        self._radius = radius
+        self._lower = lower
+        self._upper = upper
+        self._topography = None
+
+    @property
+    def topography(self):
+        if "cf_units" not in sys.modules:
+            raise ModuleNotFoundError(
+                "valleyfloorrelativealtitudefilter filter is missing required dependency 'cf-units'. Please install to use this filter."
+            )
+        if "xarray" not in sys.modules:
+            raise ModuleNotFoundError(
+                "valleyfloorrelativealtitudefilter filter is missing required dependency 'xarray'. Please install to use this filter."
+            )
+
+        if self._topography is None:
+            try:
+                with xr.open_dataset(self._topo_file) as topo:
+                    topo = topo.fillna(0)
+                    self._topography = topo
+            except Exception as ex:
+                raise FilterException(
+                    f"Cannot read topography from '{self._topo_file}:{self._topo_var}' : {ex}"
+                )
+        return self._topography
+
+    def init_kwargs(self):
+        return {
+            "topo_file": self._topo_file,
+            "topo_var": self._topo_var,
+            "radius": self._radius,
+            "lower": self._lower,
+            "upper": self._upper,
+        }
+
+    def name(self):
+        return "valleyfloorrelativealtitudefilter"
+
+    def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+        filtered_stations = {}
+        for k, v in stations.items():
+            lat = v.latitude
+            lon = v.longitude
+            alt = v.altitude
+
+            ralt = self._calculate_relative_altitude(
```

Comment:
Only here, the topography data is needed. Open right before, close right after this function?